osidb/serializer.py

```diff
@@ -287,18 +289,65 @@ class Meta:
         ] + TrackingMixinSerializer.Meta.fields
 
 
-class MetaSerializer(TrackingMixinSerializer):
+@extend_schema_field(
+    {
+        "type": "array",
+        "items": {"type": "string", "minLength": 1},
+        "minItems": 1,
+        "uniqueItems": True,
+    }
+)
+class ACLSerializer(serializers.ListField):
```

Comment:
What's the rationale behind letting API users set ACLs? I don't think they should be able to or at least not directly, right now we should only care about setting it for them based on whether a flaw is embargoed or not

Reply:
In the future we might want to let them add/remove groups that can view/edit a flaw/etc. but not completely override `acl_read` and `acl_write` directly, I consider those as backend-only fields.

Reply:
Sure that now we could make it with public/embargoed but this is not future proof. Later there can be arbitrary groups being probably dynamically changed by the users in some kind of product definitions (product/component related groups etc.).

What do you see in between capability of add/remove groups that can view/edit a flaw/etc. and completely overriding the ACLs? I do not see a difference.

I see the responsibility of the backend to ensure that the user only assigns the groups (s)he is allowed to to the entities (s)he is allowed to. This is my motivation to make the API very general but introduce the ACL validations.

Reply:
I just think that something like encapsulation works better in this particular case, if we ever need to change the `acl_read`, `acl_write` fields we can do so without breaking API, by providing an interface or abstraction at the API level to add/remove only a subset of groups we guarantee that we can change the backend without affecting the API.

Reply:
So how would you rather see this implemented? Something like public/embargoed then extended to eg. some team/product terminology? The user needs to somehow specify the access/visibility and if it is not directly then indirectly through some mapping - the mentioned abstraction.

Currently I can imagine abstracting to something like embargoed boolean field. However that would be definitely broken in the future. We do not know much about what will come so I am not sure how to abstract it. Would it be like embargoed+OCP+RHEL? How would we define read vs write? To many unknowns for me. Therefore I came with not abstracting.

I am not sure how should this approach lead to breaking API. That would require that we reimplement the backend ACL mechanism so it is no more based on `acl_read` and `acl_write` which is probably big enough change worth breaking anything. We do not plan any such change.

Reply:
> So how would you rather see this implemented? 

At the API level, we would have a single `groups` API-only field which would serve as the abstraction, a response with this field could look something like:

```json
{
    "groups": {
        "redhat": {"read": true, "write": false}
    }
}
```

When reading, the field would **not** include any of the various security-related groups, such as embargo read/write, osidb-admin groups, etc.

When writing, the field should refuse/ignore whenever a client passes any of the security-related groups that I mentioned previously, but accept any other valid LDAP group (whether the group is valid or not is up to the client), instead those security-related groups should **only** be set via a different way (like an embargo boolean).

I don't want clients to be able to set the embargo/public/admin groups or even read them directly because **they don't need to**, even if we have validations that prevent those things from happening bugs also can happen and the risk of letting them set `acl_read` and `acl_write` directly just isn't worth it.

> I am not sure how should this approach lead to breaking API.

It doesn't, but by having encapsulation we can prevent API breakage in the future, if we ever completely change the ACL system we can keep the public API as a list/object of ldap group names and we can change the backend however we want. If we just let users directly modify `acl_read` and `acl_write` then if we do decide to change anything in the future it might be more painful.

That said that's just my opinion, there's nothing technically wrong with the implementation just not what I'd prefer from a design pov.

Reply:
@Elkasitu the hard thing is that the future usage of the groups is not very clear. So far we have public or embargoed visibility and that is it. If we had embargoed plus this and that team is it still embargoed?

What would you say on the following temporary solution? I will refactor the UI to accept only the embargoed boolean for now - no ACL groups. The serializer would translate it to the corresponding ACLs. For now there would basically be two possible sets (all public read and write or all embargoed read and write groups). We would deal with the higher granularity later when we know more.

Reply:
I reworked the PR so that now the API accepts `embargoed` attribute (and is mandatory) and sets the ACLs appropriately. No other options/granularity are allowed now - will be dealt with when we know more.